core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -192,8 +212,11 @@ public SnapshotInfo snapshot(SnapshotId snapshotId) {
      * @param listener snapshot creation listener
      */
     public void createSnapshot(final SnapshotRequest request, final CreateSnapshotListener listener) {
-        final SnapshotId snapshotId = new SnapshotId(request.repository(), request.name());
-        validate(snapshotId);
+        final String repositoryName = request.repositoryName;
+        final String snapshotName = request.snapshotName;
+        validate(repositoryName, snapshotName);
+        final SnapshotId snapshotId = new SnapshotId(snapshotName, UUIDs.randomBase64UUID()); // new UUID for the snapshot
```

Comment:
We call `UUIDs.randomBase64UUID()` in 17 different places in this PR. Maybe move into a SnapshotId constructor or a static create(String name) method?